File: ocaml/99/14.ml
(* Duplicate the elements of a list. (easy)

# duplicate ["a";"b";"c";"c";"d"];;
- : string list = ["a"; "a"; "b"; "b"; "c"; "c"; "c"; "c"; "d"; "d"]
 *)

let rec duplicate = function
  | [] -> []
  | x :: t -> x :: x :: duplicate t ;;

  

File: prolog/99problems/14.pl
/* Duplicate the elements of a list.
Example:
?- dupli([a,b,c,c,d],X).
X = [a,a,b,b,c,c,c,c,d,d]
*/

dupli([],[]).
dupli([X|T], [X,X|S]) :- dupli(T,S).
